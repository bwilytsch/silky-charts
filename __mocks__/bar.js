/* ignore file coverage */

export const data = [
  { name: 'foo', value: 19 },
  { name: 'bar', value: 70 },
  { name: 'baz', value: 93 },
  { name: 'quz', value: 80 },
];
export const dataWidthDates = [
  { name: '2018-01-31T05:00:00.000Z', value: 19 },
  { name: '2018-02-01T05:00:00.000Z', value: 70 },
  { name: '2018-02-02T05:00:00.000Z', value: 93 },
  { name: '2018-02-03T05:00:00.000Z', value: 80 },
];
