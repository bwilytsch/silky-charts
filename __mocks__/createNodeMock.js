export default element => {
  if (element.type === 'svg') {
    return {
      parentElement: {},
    };
  }
};
