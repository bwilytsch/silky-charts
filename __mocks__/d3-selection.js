/* ignore file coverage */

const selection = jest.genMockFromModule('d3-selection');

const attr = () => ({ attr });

const fn = () => ({
  call: jest.fn(),
  select: jest.fn(() => ({ attr })),
  selectAll: jest.fn(() => ({ attr })),
});

selection.select = fn;
selection.selectAll = fn;

module.exports = selection;
