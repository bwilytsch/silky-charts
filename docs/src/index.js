import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const baseURL = '/silky-charts';
  const pathname = window.location.pathname;
  const currentPath = pathname.includes(baseURL)
    ? pathname.replace(baseURL, '')
    : '/';
  const isValidPathname =
    currentPath && pages.map(({ path }) => path).includes(currentPath);

  console.log(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        baseURL={baseURL}
        initial={isValidPathname ? currentPath : '/'}
      />
      <Router primary={false}>
        {currentPath === '/' && (
          <Redirect from={currentPath} to={baseURL} noThrow />
        )}
        <Content path={baseURL}>
          {pages.map(({ component, path }, idx) => {
            const Component = contentComponents[component];
            return <Component key={idx} path={path} default={path === '/'} />;
          })}
        </Content>
      </Router>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
