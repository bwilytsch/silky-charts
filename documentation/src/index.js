import React from 'react';
import { render } from 'react-dom';
import { Router, Redirect } from '@reach/router';
import Layout from './containers/Layout';
import Header from './containers/Header';
import Navigation from './containers/Navigation';
import Content from './containers/Content';
import Introduction from './components/Introduction';
import BarChart from './components/BarChart';
import pages from './pages.json';
import 'normalize.css';
import './styles.css';

const App = () => {
  const pathname = window.location.pathname;
  const defaultPath = '/';
  const currentPath =
    pathname !== defaultPath ? pathname.replace(defaultPath, '') : pathname;
  const isInvalidPath = !pages
    .map(({ pathname }) => pathname)
    .includes(currentPath);

  const contentComponents = {
    Introduction,
    BarChart,
  };

  return (
    <Layout>
      <Header />
      <Navigation
        items={pages}
        initialPath={isInvalidPath ? defaultPath : currentPath}
      />
      <Content>
        <Router>
          {isInvalidPath && (
            <Redirect from={currentPath} to={defaultPath} noThrow />
          )}
          {pages.map(({ component, pathname }, idx) => {
            const Component = contentComponents[component];
            return (
              <Component
                key={idx}
                path={pathname}
                default={pathname === defaultPath}
              />
            );
          })}
        </Router>
      </Content>
    </Layout>
  );
};

render(<App />, document.getElementById('app'));
