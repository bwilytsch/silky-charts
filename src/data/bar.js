export default [
  {
    name: '2018-02-01T05:00:00.000Z',
    value: 70,
  },
  {
    name: '2018-02-02T05:00:00.000Z',
    value: 93,
  },
  {
    name: '2018-02-03T05:00:00.000Z',
    value: 80,
  },
  {
    name: '2018-02-04T05:00:00.000Z',
    value: 99,
  },
  {
    name: '2018-02-05T05:00:00.000Z',
    value: 31,
  },
  {
    name: '2018-02-06T05:00:00.000Z',
    value: 23,
  },
  {
    name: '2018-02-07T05:00:00.000Z',
    value: 13,
  },
  {
    name: '2018-02-08T05:00:00.000Z',
    value: 81,
  },
  {
    name: '2018-02-09T05:00:00.000Z',
    value: 19,
  },
  {
    name: '2018-02-10T05:00:00.000Z',
    value: 75,
  },
  {
    name: '2018-02-11T05:00:00.000Z',
    value: 86,
  },
  {
    name: '2018-02-12T05:00:00.000Z',
    value: 91,
  },
  {
    name: '2018-02-13T05:00:00.000Z',
    value: 63,
  },
  {
    name: '2018-02-14T05:00:00.000Z',
    value: 28,
  },
  {
    name: '2018-02-15T05:00:00.000Z',
    value: 43,
  },
  {
    name: '2018-02-16T05:00:00.000Z',
    value: 88,
  },
  {
    name: '2018-02-17T05:00:00.000Z',
    value: 29,
  },
  {
    name: '2018-02-18T05:00:00.000Z',
    value: 9,
  },
  {
    name: '2018-02-19T05:00:00.000Z',
    value: 29,
  },
  {
    name: '2018-02-20T05:00:00.000Z',
    value: 73,
  },
  {
    name: '2018-02-21T05:00:00.000Z',
    value: 2,
  },
  {
    name: '2018-02-22T05:00:00.000Z',
    value: 61,
  },
  {
    name: '2018-02-23T05:00:00.000Z',
    value: 36,
  },
  {
    name: '2018-02-24T05:00:00.000Z',
    value: 8,
  },
  {
    name: '2018-02-25T05:00:00.000Z',
    value: 78,
  },
  {
    name: '2018-02-26T05:00:00.000Z',
    value: 20,
  },
  {
    name: '2018-02-27T05:00:00.000Z',
    value: 24,
  },
  {
    name: '2018-02-28T05:00:00.000Z',
    value: 97,
  },
];
