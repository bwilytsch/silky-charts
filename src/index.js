export { default as Bar } from './Bar';
export { default as BarLine } from './BarLine';
export { default as StackedArea } from './StackedArea';

export { default as toSeries } from './utils/toSeries';
