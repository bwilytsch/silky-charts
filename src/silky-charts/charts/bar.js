// @flow
import type { Bar } from '../types'
import { select, selectAll, selection } from 'd3-selection' // eslint-disable-line
import { axisBottom, axisLeft } from 'd3-axis'
import { areAllDate } from '../utils'
import {
  drawGrid,
  drawLabels,
  drawReferenceLine,
  getXScale,
  getYScale,
  highValues,
} from './helpers'
import 'd3-transition'

export default ({
  currency,
  currencyshortNotation,
  data,
  grid,
  horizontal,
  localization,
  margin = { top: 60, right: 60, bottom: 60, left: 60 },
  ref,
  referenceLine,
  shortNotation,
  size,
  source,
  title,
  valueOnBars,
  xAxisLabel,
  yAxisLabel,
}: Bar) => {
  const svg = select(ref)
  const height = size.height - margin.top - margin.bottom
  const width = size.width - margin.left - margin.right
  const drawLabel = drawLabels(svg, size, margin)
  // const areNamesDate = areAllDate(data.map(d => d.name))
  const areNamesDate = false
  const isValueHidden = x => height - yScale(x) < 50

  const chart = svg
    .append('g')
    .attr('transform', `translate(${margin.left}, ${margin.top})`)

  const xScale = getXScale(areNamesDate ? 'time' : 'band', data, width)
  const yScale = getYScale('linear', data, height)

  const xAxis = chart
    .append('g')
    .attr('class', 'x-axis')
    .attr('transform', `translate(0, ${height})`)
    .call(axisBottom(xScale))

  const yAxis = chart
    .append('g')
    .attr('class', 'y-axis')
    .call(axisLeft(yScale))

  // Translate and rotate the x axis labels when they are not instance of Date
  if (!areNamesDate) {
    xAxis
      .selectAll('text')
      .attr('text-anchor', 'end')
      .attr('transform', 'translate(-12, 6) rotate(-50)')
  }

  // Extend x axis path length if the names values are instance of Date
  if (areNamesDate) {
    chart.select('.x-axis path.domain').attr('d', () => `M0,0.5V0.5H${width}`)
  }

  if (grid) {
    chart
      .append('g')
      .attr('class', 'grid')
      .call(drawGrid(horizontal, xScale, height, yScale, width))
  }

  const barGroup = chart
    .append('g')
    .selectAll()
    .data(data)
    .enter()
    .append('g')

  barGroup
    .append('rect')
    .attr('class', 'bar')
    .attr('x', ({ name }) =>
      areNamesDate
        ? xScale(new Date(name)) - width / data.length / 2.65
        : xScale(name)
    )
    .attr('y', ({ value }) => yScale(value))
    .attr('height', ({ value }) => height - yScale(value))
    .attr(
      'width',
      areNamesDate ? width / (data.length * 1.25) : xScale.bandwidth()
    )

  if (valueOnBars && !highValues(data)) {
    barGroup
      .append('text')
      .attr('class', ({ value }) =>
        isValueHidden(value) ? 'value out' : 'value'
      )
      .attr('transform', ({ name, value }) => {
        const x = areNamesDate
          ? xScale(new Date(name))
          : xScale(name) + xScale.bandwidth() / 2
        const y = isValueHidden(value) ? yScale(value) - 20 : yScale(value) + 30
        return `translate(${x}, ${y})`
      })
      .attr('text-anchor', 'middle')
      .text(({ value }) => value)
  }

  barGroup
    .selectAll('.bar')
    .on('mouseenter', function({ value: currentValue }) {
      referenceLine && drawReferenceLine(chart, yScale(currentValue), width)

      if (valueOnBars) {
        const _parentNode = this.parentNode
        selectAll('.value').text(function({ value }) {
          const divergence = value - currentValue
          const text = divergence > 0 ? `+${divergence}` : `${divergence}`
          return _parentNode === this.parentNode ? '' : text
        })
      }
    })
    .on('mouseleave', function() {
      valueOnBars && selectAll('.value').text(({ value }) => value)
      referenceLine && chart.select('.reference-line').remove()
    })

  if (title) {
    drawLabel('title', title)
  }

  if (xAxisLabel) {
    drawLabel('xAxis', xAxisLabel)
  }

  if (yAxisLabel) {
    drawLabel('yAxis', yAxisLabel)
  }

  if (source) {
    drawLabel('source', source)
  }

  return chart
}
