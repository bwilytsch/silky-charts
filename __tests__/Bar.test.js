import React from 'react';
import { Bar } from '../src';
import { render, cleanup } from 'react-testing-library';
import { data, dataWidthDates } from '../__mocks__/bar';
import 'jest-styled-components';

afterEach(cleanup);

test('Bar', () => {
  const { container } = render(<Bar data={data} />);

  const bar = container.firstChild;
  expect(bar).toMatchSnapshot();
});

test('Bar with dates', () => {
  const { container } = render(<Bar data={dataWidthDates} />);

  const bar = container.firstChild;
  expect(bar).toMatchSnapshot();
});

test('Bar responsive', () => {
  const { container } = render(<Bar data={data} responsive />);

  const bar = container.firstChild;
  expect(bar).toMatchSnapshot();
});

test('Bar with width', () => {
  const { container } = render(<Bar data={data} width={100} />);

  const bar = container.firstChild;
  expect(bar).toMatchSnapshot();
});

test('Bar with height', () => {
  const { container } = render(<Bar data={data} height={100} />);

  const bar = container.firstChild;
  expect(bar).toMatchSnapshot();
});

test('Bar width grid', () => {
  const { container } = render(<Bar data={data} grid />);

  const bar = container.firstChild;
  expect(bar).toMatchSnapshot();
});

test('Bar with Title, X label, Y label', () => {
  const { container } = render(
    <Bar
      data={data}
      title={'Test title'}
      xAxisChartLabel={'Test X label'}
      yAxisChartLabel={'Test Y label'}
    />
  );

  const bar = container.firstChild;
  expect(bar).toMatchSnapshot();
});

test('Bar with source', () => {
  const { container } = render(<Bar data={data} dataSource="Test source" />);

  const bar = container.firstChild;
  expect(bar).toMatchSnapshot();
});

test('Bar with source link', () => {
  const { container } = render(
    <Bar data={data} dataSource={{ href: '#', target: 'foo', title: 'bar' }} />
  );

  const bar = container.firstChild;
  expect(bar).toMatchSnapshot();
});

test('Bar with X label rotation', () => {
  const { container } = render(<Bar data={data} xAxisLabelRotation />);

  const bar = container.firstChild;
  expect(bar).toMatchSnapshot();
});
