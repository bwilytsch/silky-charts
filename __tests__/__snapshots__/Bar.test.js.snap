// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar 1`] = `
.c1 {
  fill: rgb(8,104,172);
}

.c1:hover {
  fill: rgb(0,84,152);
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="dataviz-layer "
    >
      <rect
        class="bar c1"
        height="55.16129032258064"
        width="110.6024096385542"
        x="19.518072289156635"
        y="214.83870967741936"
      />
      <rect
        class="bar c1"
        height="203.2258064516129"
        width="110.6024096385542"
        x="149.63855421686748"
        y="66.7741935483871"
      />
      <rect
        class="bar c1"
        height="270"
        width="110.6024096385542"
        x="279.7590361445783"
        y="0"
      />
      <rect
        class="bar c1"
        height="232.25806451612902"
        width="110.6024096385542"
        x="409.87951807228916"
        y="37.741935483870975"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`Bar responsive 1`] = `
.c1 {
  fill: rgb(8,104,172);
}

.c1:hover {
  fill: rgb(0,84,152);
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="dataviz-layer "
    >
      <rect
        class="bar c1"
        height="55.16129032258064"
        width="110.6024096385542"
        x="19.518072289156635"
        y="214.83870967741936"
      />
      <rect
        class="bar c1"
        height="203.2258064516129"
        width="110.6024096385542"
        x="149.63855421686748"
        y="66.7741935483871"
      />
      <rect
        class="bar c1"
        height="270"
        width="110.6024096385542"
        x="279.7590361445783"
        y="0"
      />
      <rect
        class="bar c1"
        height="232.25806451612902"
        width="110.6024096385542"
        x="409.87951807228916"
        y="37.741935483870975"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`Bar width grid 1`] = `
.c1 path {
  stroke: transparent;
}

.c1 line {
  stroke: rgb(243,243,243);
}

.c2 {
  fill: rgb(8,104,172);
}

.c2:hover {
  fill: rgb(0,84,152);
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="grid c1"
    />
    <g
      class="dataviz-layer "
    >
      <rect
        class="bar c2"
        height="55.16129032258064"
        width="110.6024096385542"
        x="19.518072289156635"
        y="214.83870967741936"
      />
      <rect
        class="bar c2"
        height="203.2258064516129"
        width="110.6024096385542"
        x="149.63855421686748"
        y="66.7741935483871"
      />
      <rect
        class="bar c2"
        height="270"
        width="110.6024096385542"
        x="279.7590361445783"
        y="0"
      />
      <rect
        class="bar c2"
        height="232.25806451612902"
        width="110.6024096385542"
        x="409.87951807228916"
        y="37.741935483870975"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`Bar with Title, X label, Y label 1`] = `
.c2 {
  font-size: 0.9em;
  text-anchor: middle;
}

.c3 {
  font-size: 0.9em;
  text-anchor: middle;
  -webkit-transform: rotate(-90deg);
  -ms-transform: rotate(-90deg);
  transform: rotate(-90deg);
}

.c4 {
  fill: rgb(8,104,172);
}

.c4:hover {
  fill: rgb(0,84,152);
}

.c0 {
  height: 360px;
  width: 640px;
}

.c1 {
  font-size: 1.2em;
  text-anchor: middle;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <text
      class="chart-title c1"
      height="270"
      width="540"
      x="270"
      y="10"
    >
      Test title
    </text>
    <text
      class="x-axis-label c2"
      height="270"
      width="540"
      x="270"
      y="290"
    >
      Test X label
    </text>
    <text
      class="y-axis-label c3"
      height="270"
      width="540"
      x="-135"
      y="-16"
    >
      Test Y label
    </text>
    <g
      class="dataviz-layer "
    >
      <rect
        class="bar c4"
        height="55.16129032258064"
        width="110.6024096385542"
        x="19.518072289156635"
        y="214.83870967741936"
      />
      <rect
        class="bar c4"
        height="203.2258064516129"
        width="110.6024096385542"
        x="149.63855421686748"
        y="66.7741935483871"
      />
      <rect
        class="bar c4"
        height="270"
        width="110.6024096385542"
        x="279.7590361445783"
        y="0"
      />
      <rect
        class="bar c4"
        height="232.25806451612902"
        width="110.6024096385542"
        x="409.87951807228916"
        y="37.741935483870975"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`Bar with X label rotation 1`] = `
.c1 {
  fill: rgb(8,104,172);
}

.c1:hover {
  fill: rgb(0,84,152);
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="dataviz-layer "
    >
      <rect
        class="bar c1"
        height="55.16129032258064"
        width="110.6024096385542"
        x="19.518072289156635"
        y="214.83870967741936"
      />
      <rect
        class="bar c1"
        height="203.2258064516129"
        width="110.6024096385542"
        x="149.63855421686748"
        y="66.7741935483871"
      />
      <rect
        class="bar c1"
        height="270"
        width="110.6024096385542"
        x="279.7590361445783"
        y="0"
      />
      <rect
        class="bar c1"
        height="232.25806451612902"
        width="110.6024096385542"
        x="409.87951807228916"
        y="37.741935483870975"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`Bar with dates 1`] = `
.c1 {
  fill: rgb(8,104,172);
}

.c1:hover {
  fill: rgb(0,84,152);
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="dataviz-layer "
    >
      <rect
        class="bar c1"
        height="55.16129032258064"
        width="110.6024096385542"
        x="19.518072289156635"
        y="214.83870967741936"
      />
      <rect
        class="bar c1"
        height="203.2258064516129"
        width="110.6024096385542"
        x="149.63855421686748"
        y="66.7741935483871"
      />
      <rect
        class="bar c1"
        height="270"
        width="110.6024096385542"
        x="279.7590361445783"
        y="0"
      />
      <rect
        class="bar c1"
        height="232.25806451612902"
        width="110.6024096385542"
        x="409.87951807228916"
        y="37.741935483870975"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`Bar with height 1`] = `
.c1 {
  fill: rgb(8,104,172);
}

.c1:hover {
  fill: rgb(0,84,152);
}

.c0 {
  height: 100px;
  width: 177.77777777777777px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="dataviz-layer "
    >
      <rect
        class="bar c1"
        height="2.043010752688172"
        width="15.930388219544845"
        x="2.811244979919678"
        y="7.956989247311828"
      />
      <rect
        class="bar c1"
        height="7.526881720430108"
        width="15.930388219544845"
        x="21.5528781793842"
        y="2.473118279569892"
      />
      <rect
        class="bar c1"
        height="10"
        width="15.930388219544845"
        x="40.294511378848725"
        y="0"
      />
      <rect
        class="bar c1"
        height="8.602150537634408"
        width="15.930388219544845"
        x="59.03614457831325"
        y="1.397849462365592"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 10)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`Bar with source 1`] = `
.c2 {
  fill: rgb(8,104,172);
}

.c2:hover {
  fill: rgb(0,84,152);
}

.c1 {
  font-size: 0.7em;
  font-style: italic;
  text-anchor: end;
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <text
      class="data-source c1"
      height="270"
      width="540"
      x="540"
      y="270"
    >
      Test source
    </text>
    <g
      class="dataviz-layer "
    >
      <rect
        class="bar c2"
        height="55.16129032258064"
        width="110.6024096385542"
        x="19.518072289156635"
        y="214.83870967741936"
      />
      <rect
        class="bar c2"
        height="203.2258064516129"
        width="110.6024096385542"
        x="149.63855421686748"
        y="66.7741935483871"
      />
      <rect
        class="bar c2"
        height="270"
        width="110.6024096385542"
        x="279.7590361445783"
        y="0"
      />
      <rect
        class="bar c2"
        height="232.25806451612902"
        width="110.6024096385542"
        x="409.87951807228916"
        y="37.741935483870975"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`Bar with source link 1`] = `
.c3 {
  fill: rgb(8,104,172);
}

.c3:hover {
  fill: rgb(0,84,152);
}

.c1 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 {
  font-size: 0.7em;
  font-style: italic;
  text-anchor: end;
}

.c0 {
  height: 360px;
  width: 640px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <a
      class="data-source-link c1"
      href="#"
      target="foo"
      title="bar"
    >
      <text
        class="data-source c2"
        height="270"
        width="540"
        x="540"
        y="270"
      />
    </a>
    <g
      class="dataviz-layer "
    >
      <rect
        class="bar c3"
        height="55.16129032258064"
        width="110.6024096385542"
        x="19.518072289156635"
        y="214.83870967741936"
      />
      <rect
        class="bar c3"
        height="203.2258064516129"
        width="110.6024096385542"
        x="149.63855421686748"
        y="66.7741935483871"
      />
      <rect
        class="bar c3"
        height="270"
        width="110.6024096385542"
        x="279.7590361445783"
        y="0"
      />
      <rect
        class="bar c3"
        height="232.25806451612902"
        width="110.6024096385542"
        x="409.87951807228916"
        y="37.741935483870975"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, 270)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;

exports[`Bar with width 1`] = `
.c1 {
  fill: rgb(8,104,172);
}

.c1:hover {
  fill: rgb(0,84,152);
}

.c0 {
  height: 56.25px;
  width: 100px;
}

<svg
  baseProfile="full"
  class="silky-charts c0"
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    class="silky-charts-container "
    transform="translate(50, 40)"
  >
    <g
      class="dataviz-layer "
    >
      <rect
        class="bar c1"
        height="-6.89516129032258"
        width="0"
        x="0"
        y="-26.85483870967742"
      />
      <rect
        class="bar c1"
        height="-25.403225806451612"
        width="0"
        x="0"
        y="-8.346774193548388"
      />
      <rect
        class="bar c1"
        height="-33.75"
        width="0"
        x="0"
        y="0"
      />
      <rect
        class="bar c1"
        height="-29.032258064516128"
        width="0"
        x="0"
        y="-4.717741935483872"
      />
    </g>
    <g
      class="axis-x "
      transform="translate(0, -33.75)"
    />
    <g
      class="axis-y "
    />
  </g>
</svg>
`;
